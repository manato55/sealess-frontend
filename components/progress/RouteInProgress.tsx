import React, { useEffect, useState } from 'react'
import {useProgress} from '../../hooks/useProgress'
import Common from '../../styles/Common.module.scss'
import styled from 'styled-components'

type basicProps = {
    taskRoute: any;
    completed?: boolean;
}

export const RouteInProgress = (props: basicProps): React.ReactElement => {
    const [pplInRoute, setPplInRoute] = useState<string[]>([]);


    useEffect(() => {
        if(props.taskRoute.length > 0) {
            let keys: string[] = Object.keys(props.taskRoute[0])
            let routeExtracted: string[] = keys.filter(v => v.match(/route[1-5]_user/))
            setPplInRoute(routeExtracted)
        }
    }, [props.taskRoute])


    return (
        <div className={Common.input_container}>
            {pplInRoute.length > 0 ? pplInRoute.map((routeNum, index) => 
                <div className={Common.route_container} key={index}>
                    {props.taskRoute[0][routeNum] !== null && 
                        <div>
                            {index !== 0 && <p>↓</p>}
                            <div className={Common.ppl_involved}>
                                <p>
                                    <Involved>関与者{index+1}</Involved>
                                    {!props.completed && routeNum.slice(0,6) === props.taskRoute[0].process && 
                                        <Holder>案件保持</Holder>
                                    }
                                </p>
                                {props.taskRoute[0][routeNum].department}&emsp;{props.taskRoute[0][routeNum].section}&emsp;{props.taskRoute[0][routeNum].name}
                            </div>
                        </div>
                        
                    }
                </div>
            ):''}
        </div>
    )
}

const Involved = styled.span`
    margin-right: 20px;
`;

const Holder = styled.span`
    color: red;
`;


export default RouteInProgress