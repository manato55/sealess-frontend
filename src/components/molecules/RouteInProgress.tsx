import React, { useEffect, useState } from 'react';
import styled from 'styled-components';

interface Props {
  taskRoute: any;
  completed?: boolean;
}

export const RouteInProgress = (props: Props): React.ReactElement => {
  const [pplInRoute, setPplInRoute] = useState<string[]>([]);
  let cnt = 0;

  const countUp = () => {
    cnt++;
  };

  const countRestore = () => {
    cnt = 0;
  };

  useEffect(() => {
    if (props.taskRoute.length > 0) {
      let keys: string[] = Object.keys(props.taskRoute[0]);
      let routeExtracted: string[] = keys.filter((v) => v.match(/route[1-5]_user/));
      setPplInRoute(routeExtracted);
    }
  }, [props.taskRoute]);

  return (
    <InputContainer>
      {pplInRoute.length > 0 &&
        pplInRoute.map((routeNum, index) => (
          <RouteContainer key={index}>
            {props.taskRoute[0][routeNum] !== null && (
              <div>
                {countRestore()}
                {index !== 0 && <p>↓</p>}
                <PplInvolved>
                  <p>
                    <Involved>関与者{index + 1}</Involved>
                    {!props.completed && routeNum.slice(0, 6) === props.taskRoute[0].process && (
                      <Holder>案件保持</Holder>
                    )}
                  </p>
                  {props.taskRoute[0].agent_statuses?.map((agent, v) => (
                    <div key={v}>
                      {`${agent.route}_user` === routeNum && (
                        <p>
                          {countUp()}
                          {agent.user.department.name}&emsp;{agent.user.section.name}&emsp;
                          {agent.user.name}
                        </p>
                      )}
                    </div>
                  ))}
                  {cnt > 0 && props.taskRoute[0][routeNum] !== null ? (
                    <Highlight>
                      {`（代理：${props.taskRoute[0][routeNum].department.name}`}&emsp;
                      {props.taskRoute[0][routeNum].section.name}&emsp;
                      {`${props.taskRoute[0][routeNum].name}）`}
                    </Highlight>
                  ) : (
                    <span>
                      {props.taskRoute[0][routeNum].deleted_at ? (
                        <span>
                          <Highlight>削除ユーザー</Highlight>(
                          {props.taskRoute[0][routeNum].department.name}&emsp;
                          {props.taskRoute[0][routeNum].section.name}&emsp;
                          {props.taskRoute[0][routeNum].name})
                        </span>
                      ) : (
                        <span>
                          {props.taskRoute[0][routeNum].department.name}&emsp;
                          {props.taskRoute[0][routeNum].section.name}&emsp;
                          {props.taskRoute[0][routeNum].name}
                        </span>
                      )}
                    </span>
                  )}
                </PplInvolved>
              </div>
            )}
          </RouteContainer>
        ))}
    </InputContainer>
  );
};

const InputContainer = styled.div`
  width: 90%;
  margin: 30px auto;
  background: gainsboro;
  padding: 30px;
  border-radius: 10px;
`;

const Involved = styled.span`
  margin-right: 20px;
`;

const Holder = styled.span`
  color: red;
`;

const Highlight = styled.p`
  color: red;
`;

const RouteContainer = styled.div`
  text-align: center;
  margin-top: 20px;
`;

const PplInvolved = styled.div`
  width: 80%;
  margin: 0 auto;
  background: white;
  border-radius: 10px;
`;

export default RouteInProgress;
