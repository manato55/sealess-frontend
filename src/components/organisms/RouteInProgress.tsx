import React, { useEffect, useState } from 'react';
import styled from 'styled-components';

type basicProps = {
  taskRoute: any;
  completed?: boolean;
};

export const RouteInProgress = (props: basicProps): React.ReactElement => {
  const [pplInRoute, setPplInRoute] = useState<string[]>([]);
  let cnt: number = 0;

  const countUp = () => {
    cnt++;
  };

  const countRestore = () => {
    cnt = 0;
  };

  useEffect(() => {
    if (props.taskRoute.length > 0) {
      let keys: string[] = Object.keys(props.taskRoute[0]);
      let routeExtracted: string[] = keys.filter((v) => v.match(/route[1-5]_user/));
      setPplInRoute(routeExtracted);
    }
  }, [props.taskRoute]);

  return (
    <InputContainer>
      {pplInRoute.length > 0 &&
        pplInRoute.map((routeNum, index) => (
          <RouteContainer key={index}>
            {props.taskRoute[0][routeNum] !== null && (
              <div>
                {countRestore()}
                {index !== 0 && <p>↓</p>}
                <PplInvolved>
                  <p>
                    <Involved>関与者{index + 1}</Involved>
                    {!props.completed && routeNum.slice(0, 6) === props.taskRoute[0].process && (
                      <Holder>案件保持</Holder>
                    )}
                  </p>
                  {props.taskRoute[0].agent_statuses?.map((agent, v) => (
                    <div key={v}>
                      {`${agent.route}_user` === routeNum && (
                        <p>
                          {countUp()}
                          {agent.user.department}&emsp;{agent.user.section}&emsp;{agent.user.name}
                        </p>
                      )}
                    </div>
                  ))}
                  {cnt > 0 ? (
                    <AgentPerson>
                      {`（代理：${props.taskRoute[0][routeNum].department}`}&emsp;
                      {props.taskRoute[0][routeNum].section}&emsp;
                      {`${props.taskRoute[0][routeNum].name}）`}
                    </AgentPerson>
                  ) : (
                    <span>
                      {props.taskRoute[0][routeNum].department}&emsp;
                      {props.taskRoute[0][routeNum].section}&emsp;
                      {props.taskRoute[0][routeNum].name}
                    </span>
                  )}
                </PplInvolved>
              </div>
            )}
          </RouteContainer>
        ))}
    </InputContainer>
  );
};

const InputContainer = styled.div`
  width: 90%;
  margin: 30px auto;
  background: gainsboro;
  padding: 30px;
  border-radius: 10px;
`;

const Involved = styled.span`
  margin-right: 20px;
`;

const Holder = styled.span`
  color: red;
`;

const AgentPerson = styled.span`
  color: red;
`;

const RouteContainer = styled.div`
  text-align: center;
  margin-top: 20px;
`;

const PplInvolved = styled.div`
  width: 80%;
  margin: 0 auto;
  background: white;
  border-radius: 10px;
`;

export default RouteInProgress;
