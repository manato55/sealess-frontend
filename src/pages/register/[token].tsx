import RegisterToken from '../../components/organisms/RegisterToken';

const Token = () => {
  return (
    <>
      <RegisterToken />
    </>
  );
};

export default Token;
