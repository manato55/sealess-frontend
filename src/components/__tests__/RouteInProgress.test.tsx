import React from 'react';
import RouteInProgress from '../molecules/RouteInProgress';
import { mount, shallow } from 'enzyme';
import 'jsdom-global/register';

describe('RouteInProgress test', () => {
  test('snapshot test', () => {
    const tree = mount(<RouteInProgress taskRoute={[]} />);
    expect(tree).toMatchSnapshot();
  });
});
