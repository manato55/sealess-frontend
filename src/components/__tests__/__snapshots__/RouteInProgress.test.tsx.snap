// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteInProgress test snapshot test 1`] = `
<RouteInProgress
  taskRoute={Array []}
>
  <styled.div>
    <div
      className="sc-bdnxRM bWftbc"
    />
  </styled.div>
</RouteInProgress>
`;
